airflow/executors/kubernetes_executor.py

```diff
@@ -380,7 +380,7 @@ def sync(self) -> None:
     def process_watcher_task(self, task: KubernetesWatchType) -> None:
         """Process the task by watcher."""
         pod_id, namespace, state, annotations, resource_version = task
-        self.log.info(
```

Comment:
I prefer this at info level as it let's you know when the pod finished to complement knowing when it started.

Reply:
The exact same thing is being logged again once the item is popped from the `result_queue` on line 598. (added to queue on line 398 right below here)

IMO the path from here to there is internal to this module and hence only deserves debug-level.

btw, the `if key` on line 387 looks redundant, can `annotations_to_key` ever return None?